### ISSUE TITLE
Simulated fog scene

### ISSUE THREAD

**Issue by MMMazart** [deborahLevy130/seathru_NeRF#7]
Can you please give the experimental parameters for defog on fern data? My own tests didn't achieve a desired result. Thank you very much.

**Comment by WHUTzmy** [comment]
Thanks for your excellent work, but I experimented with the Simulated fog scene(fern) using the experimental parameters of train_llff_uw.sh, but it didn't get the results shown in the paper. The figure below shows the experimental results ,it does not learn the backscatter, do you know what the problem is? Can you release your data
![Seathru_NeRF](https://github.com/deborahLevy130/seathru_NeRF/assets/64513315/598d2e29-504c-4670-919e-4a3b7071374b)
set and experimental parameters? appreciate it!

**Comment by deborahLevy130** [comment]
Hello,
Sorry for the late response. I added a config file for the fog simulation, you can find it [here](https://github.com/deborahLevy130/seathru_NeRF/blob/master/configs/llff_256_fog.gin). And [here](https://drive.google.com/file/d/1NhqomTVReQClOQis5cTbsQkoSgODnFcJ/view?usp=sharing) are the images I used.

**Comment by deborahLevy130** [comment]
Hi,
I apologize for the late reply. I had to check a few things before getting back to you with the answer:) 
When we submitted the paper we added samples to the backscatter component. MIP360 has a proposal network and a NeRF network, and the color/final rendering is done by the NeRF network, which samples points along the ray around the objects on the ray. This is a good thing for the object's component. Still, we thought that it was not suitable for the backscatter component because there's water everywhere along the ray, so we should have uniform sampling for the backscatter, at least until it encounters an object. This is the meaning of adding extra samples to the backscatter. We actually made an assumption that until the samples of the NeRF network, the transmittance is 1 (because there is no object).
Later on, we run a few experiments with and without the extra samples and it seemed there is no significant change for the real-world scenes. When you sent me your question I noticed that for the fog simulation it matters, so I updated the code with the option for extra samples. I also uploaded a config file for the fog simulation, and [here](https://drive.google.com/file/d/1NhqomTVReQClOQis5cTbsQkoSgODnFcJ/view?usp=drive_link) are the images I used.
By the way, you can get reasonable results without the extra samples, but not as good as with them. [here](https://github.com/deborahLevy130/seathru_NeRF/blob/master/configs/llff_fog_sim_no_extra_samp.gin) you can find a config file for the fog without extra samples.

**Comment by MMMazart** [comment]
Thank you very much for your reply, I re-git the latest version of your code, used the data and experimental parameters you provided, fetched the poses via colmap, but I still don't get the dehaze effect in the paper, to be precise, there is no effect, it doesn't decouple the scene from the haze. Is there any problem somewhere?

**Comment by deborahLevy130** [comment]
Can you please share the parameters you used?

**Comment by MMMazart** [comment]
Same parameters as you provided, I only changed one place, Config.factor = 1, which means that my training image resolution is 1008×756, which is relative to the original image's 4k resolution with factor=4. Do you have a training image resolution of 252×189? If so, that means my iteration count is low, or I need to 1/4 downsample the image.

**Comment by MMMazart** [comment]
Thank you very much for your patience in replying, it seems that my program has read in the wrong configuration file, I will try to re-run it!

**Comment by MMMazart** [comment]
I've done several experiments but only succeeded once. I compared the config.gin files they recorded and they  are the same, but the run results are different. Is it because of the random seed?

**Comment by MMMazart** [comment]
I have re-run several experiments and they can get reasonable results, but there are some gap with the results you show and I don't understand what the problem is. In my experiments I found that the learning rate affects the results significantly. For example, I set max_steps to 450000, which causes the learning rate to decay differently for each iteration, which also causes the final results to be different. I'm using a 2080ti gpu.
![reply](https://github.com/deborahLevy130/seathru_NeRF/assets/64513315/d8fccf68-3615-4cf1-bbc2-e835f2a96448)
Here are my experimental parameters, which I am sure are not wrong
# Parameters for Config:
# ==============================================================================
Config.adam_eps = 1e-08
Config.batch_size = 2048
Config.checkpoint_dir = '/home/zmy/seathru_NeRF/haze/fern_extra_sample'
Config.data_dir = '/data/zmy/fogSim//fern'
Config.data_loss_mult = 1
Config.data_loss_type = 'rawnerf'
Config.dataset_loader = 'llff'
Config.distortion_loss_mult = 0.0
Config.eval_on_train = False
Config.eval_only_once = True
Config.extra_samples = True
Config.factor = 1
Config.far = 1.0
Config.forward_facing = True
Config.gen_eq = False
Config.interlevel_loss_mult = 1
Config.lr_final = 2e-05
Config.lr_init = 0.002
Config.max_steps = 250000
Config.near = 0.0
Config.print_every = 1000
Config.train_render_every = 50000
Config.use_uw_acc_trans_loss = True
Config.use_uw_acc_weights_loss = False
Config.use_uw_mlp = True
Config.uw_acc_loss_factor = 1
Config.uw_atten_xyz = False
Config.uw_final_acc_trans_loss_mult = 1e-05
Config.uw_final_acc_weights_loss_mult = 0.0001
Config.uw_fog_model = True
Config.uw_initial_acc_trans_loss_mult = 1e-05
Config.uw_initial_acc_weights_loss_mult = 0.0001
Config.uw_old_model = False

# Parameters for Model:
# ==============================================================================
Model.num_levels = 2
Model.num_nerf_samples = 32
Model.num_prop_samples = 128
Model.opaque_background = False
Model.ray_shape = 'cylinder'

# Parameters for PropMLP:
# ==============================================================================
PropMLP.basis_shape = 'octahedron'
PropMLP.basis_subdivisions = 1
PropMLP.density_bias = -1
PropMLP.density_noise = 0
PropMLP.disable_density_normals = True
PropMLP.disable_rgb = True
PropMLP.max_deg_point = 16
PropMLP.net_depth = 4
PropMLP.net_width = 256

# Parameters for UWMLP:
# ==============================================================================
UWMLP.basis_shape = 'octahedron'
UWMLP.basis_subdivisions = 1
UWMLP.density_bias = 0
UWMLP.density_noise = 0
UWMLP.disable_density_normals = True
UWMLP.gen_eq = False
UWMLP.max_deg_point = 16
UWMLP.net_depth = 8
UWMLP.net_width = 256
UWMLP.uw_atten_xyz = False
UWMLP.uw_fog_model = True
UWMLP.uw_old_model = False
UWMLP.uw_rgb_dir = False
UWMLP.water_bias = 0

**Comment by deborahLevy130** [comment]
Hi, [here](https://campushaifaac-my.sharepoint.com/:f:/g/personal/dstein09_campus_haifa_ac_il/Ehuk4ibyf19Cg4CldqXrXXgB1k_-Gg5QYUhpTnEfaUZDLg?e=LyKGWd) you can find the simulated images with the poses and bounds we used in the paper.  Maybe the problem is from the poses you used.

**Comment by MMMazart** [comment]
Thanks for your reply, the problem has been solved！
